### PR TITLE
Feature/movement

### PR DESCRIPTION
**Pull request by marcorod94**
- Now the movement game objects is done by the transform of it.
- Collision detection is still working with bullet.

### CODE REVIEW

File: Engine/Component/ComponentCollider.cpp [hunk 1]
```diff
@@ -311,6 +313,26 @@ bool ComponentCollider::DetectCollisionWith(ComponentCollider * collider)
...
+{
+	for (auto collider : App->physics->colliders)
+	{
+		if (collider->DetectCollisionWith(this) && collider->collider_type != ComponentCollider::ColliderType::MESH && this->body->getWorldArrayIndex() != collider->body->getWorldArrayIndex()) {
+			return true;
+		}
+	}
+	return false;
+}
+
```
Comment (Dartemiss) [inline Engine/Component/ComponentCollider.cpp:326]: Here you are assuming that all colliders that are not mesh colliders are enemies, and that is a wrong assumption. For example skills like the light ball will have a sphere collider.

File: Engine/Component/ComponentCollider.cpp [hunk 2]
```diff
@@ -338,21 +360,8 @@ void ComponentCollider::SetRotationAxis()
...
-
-		btTransform trans = body->getWorldTransform();
-		btQuaternion transrot = trans.getRotation();
-
-		transrot = btQuaternion(new_rotation.x, new_rotation.y, new_rotation.z, new_rotation.w);
-		trans.setRotation(transrot);
-		body->setWorldTransform(trans);
-	}
+	velocity.x = force.x;
+	velocity.z = force.z;
```
Comment (Dartemiss) [inline Engine/Component/ComponentCollider.cpp:364]: This function is not an addforce, you are substituting a value by the parameter, for an add force it should be added.

**Review by Dartemiss — CHANGES_REQUESTED** [review]
There are a few things wrong
- Collision bugs that make you fly over a million meters [https://gyazo.com/45c83171cc2df49ad7f79bc92d6f04e5]
-Gravity is computed wrongly (plane velocity)
-Jump and double jump works bad  (it feels very weird and buggy)

I think that the approach that @mariofv said to @vidalmelero doing as Carlos recomended to us would be much better as we would have absolute control over the reaction of the player when colliding with a surface.

PD: Also I saw that if u disable physics from the collider the movement stops working, is that intended?

**Comment by mariofv** [comment]
Closing this because we have a new approach

**Pull request closed without merging**